consensus/src/lib.rs

```diff
@@ -1,23 +1,49 @@
 use serde::{Deserialize, Serialize};
 use simperby_common::{
     crypto::{Hash256, PublicKey},
-    BlockHeight, ConsensusRound, Timestamp, VotingPower,
+    ConsensusRound, PrivateKey, Timestamp, ToHash256, TypedSignature, VotingPower,
 };
 use simperby_network::{
-    dms::DistributedMessageSet as DMS,
+    dms::{DistributedMessageSet as DMS, Message, MessageFilter},
     primitives::{GossipNetwork, Storage},
     *,
 };
-use std::collections::{HashMap, HashSet};
+use std::collections::BTreeSet;
+use std::sync::Arc;
+use vetomint::*;
 
 pub type Error = anyhow::Error;
+const STATE_FILE_NAME: &str = "state.json";
+pub type Nil = ();
+const NIL_BLOCK_PROPOSAL_INDEX: BlockIdentifier = BlockIdentifier::MAX;
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct ConsensusState {
-    pub non_nil_votes: HashMap<Hash256, HashSet<PublicKey>>,
-    pub nil_votes: HashMap<Hash256, HashSet<PublicKey>>,
-    pub height: BlockHeight,
-    pub round: ConsensusRound,
+pub struct State {
+    pub consensus_state: ConsensusState,
+    /// The set of messages that have been already updated to the Vetomint state machine.
+    pub updated_message: BTreeSet<Hash256>,
+    /// The set of the block hashes that have been verified.
+    pub verified_block_hash: Vec<Hash256>,
```

Comment:
```suggestion
    pub verified_block_hashes: Vec<Hash256>,
```
Now I'm kind of confused whether it is better to use plurals..